docs/FormActions.md

```diff
@@ -0,0 +1,81 @@
+# Form Actions
+
+## Description
+
+In order to create usable forms, the components need to be able to integrate with standard Canvas App controls so that standard form actions can be executed. 
+
+The components can be hooked up to Canvas App controls via exposed custom input properties. We will focus on the **submit** and **reset** actions of our components.
+
+The behaviour and instructions described here are the same for all of the components.
+
+## Submit
+
+### Description
+
+Every component has a boolean input **Submit enabled** which dictates whether the component uses live validation (boolean=false) or on submit validation (boolean=true). 
+
+Live validation means the validation occurs as soon as the input is clicked out of. This means the errors show as soon as the **HasError** output boolean switches from false to true. For an error to show for empty field validation, a value must be entered, clicked out of, clicked back into, removed and then clicked out of again. This prevents errors showing as soon as fields are clicked into. 
+
+Submit validation means errors show on selection of a button or any other external trigger which can drive a boolean value. The **HasError** output property will switch to true as soon as an error is entered and component is clicked out of (same as live validation behaviour). However, the component error will not show to the user until the **Submit** input boolean has changed its value. The **Submit** boolean allows the errors to be triggered by an external control and every time **Submit** changes it will display the results of the validation on the component.
+
+###  Set Up
+
+Below are instructions on how to link a button up so it submits a component:
+
+1. Add a library component to a canvas app. Submit should be enabled by default
+
+2. Add a button to the app.
+
+3. On the **OnSelect** property of the button, enter the following code:
+
+```
+UpdateContext({SubmitVar: !SubmitVar});
+```
+
+Where `SubmitVar` is a generic boolean submission variable.
+
+4. On the **Submit** property of the component, enter `SubmitVar`.
+
+5. Save and refresh the application.
+
+6. In play mode, clicking the button will show the result of the component validation.
+
+### Tips
+
+- It is important to save and refresh after hooking up components in order to avoid validation failures stemming from submission variables having incorrect initial value. The variable must be false initially.
+
+- Any control that can update a variable value on an action can submit a component.
+
+- Component submission variables must be unique for submissions of single components.
```

Comment:
Don't quite understand this one?

Reply:
Its basically just the opposite of what is on the line below. I have removed anyway